### PR TITLE
As/phishing and malware warnings

### PR DESCRIPTION
**Pull request by soncuteanca**
# Description

This test verifies that when a user accesses websites suspected of phishing or malware, a warning page is displayed.

## Bugzilla bug ID

**Testrail: https://testrail.stage.mozaws.net/index.php?/cases/view/3954**
**Link: https://bugzilla.mozilla.org/show_bug.cgi?id=1905189**

## Type of change

Please delete options that are not relevant.

- [x ] New Test

# How does this resolve / make progress on that bug?
Completed

Please describe the progress or significance with respect to the bug listed above.

# Screenshots / Explanations

Please upload any relevant media or add a relevant description with respect to the bug listed above.

# Comments / Concerns

Please add a short blurb about any comments or concerns that this change might cause.

### CODE REVIEW

File: tests/security_and_privacy/test_phishing_and_malware_warnings.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+def get_error_title(driver, url, expected_title):
+    """
+    Navigate to the given URL and assert the error title text.
+    """
+    driver.get(url)
+    error_title_element = WebDriverWait(driver, 10).until(
+        EC.presence_of_element_located((By.ID, "errorTitleText"))
+    )
+    error_title = error_title_element.get_attribute("innerText")
+    assert error_title == expected_title, (
```
Comment (emilghittasv) [inline tests/security_and_privacy/test_phishing_and_malware_warnings.py:27]: We should probably refrain from performing the assertion inside helper functions. This function should probably just return the error_ttile.

File: tests/security_and_privacy/test_phishing_and_malware_warnings.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    """
+    C1901393: - This tests that when a user accesses websites suspected of phishing or malware, a warning page is
+    displayed.
+    """
+    # Create the object
+    tabs = TabBar(driver)
+
+    # Test phishing warning page
+    phishing_url = "http://www.itisatrap.org/firefox/its-a-trap.html"
+    get_error_title(driver, phishing_url, "Deceptive site ahead")
```
Comment (emilghittasv) [inline tests/security_and_privacy/test_phishing_and_malware_warnings.py:43]: The actual assertion should happen here based on the get_error_title return.

File: tests/security_and_privacy/test_phishing_and_malware_warnings.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    # Open a new tab
+    tabs.new_tab_by_button()
+    tabs.wait_for_num_tabs(2)
+    driver.switch_to.window(driver.window_handles[1])
+
+    # Test malware warning page
+    malware_url = "http://www.itisatrap.org/firefox/its-an-attack.html"
+    get_error_title(driver, malware_url, "Visiting this website may harm your computer")
```
Comment (emilghittasv) [inline tests/security_and_privacy/test_phishing_and_malware_warnings.py:52]: Same here.

**Review by emilghittasv — CHANGES_REQUESTED** [review]
This looks good but we should perform the assertion inside the actual test (not inside the helper function).

File: tests/security_and_privacy/test_phishing_and_malware_warnings.py (hunk 1 above)
Comment (soncuteanca) [reply tests/security_and_privacy/test_phishing_and_malware_warnings.py]: Ok, good to know. I've moved the assertion inside the actual test.

File: tests/security_and_privacy/test_phishing_and_malware_warnings.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+def test_phishing_and_malware_warning_errors(driver: Firefox):
+    """
+    C1901393: - This tests that when a user accesses websites suspected of phishing or malware, a warning page is
+    displayed.
+    """
+    # Create the object
+    tabs = TabBar(driver)
+
+    # Test phishing warning page
+    phishing_url = "http://www.itisatrap.org/firefox/its-a-trap.html"
```
Comment (sarinali) [inline tests/security_and_privacy/test_phishing_and_malware_warnings.py:38]: Can we move constants like URLs to the top in the format of `PHISHING_URL = "http://www.itisatrap.org/firefox/its-a-trap.html"`?

File: tests/security_and_privacy/test_phishing_and_malware_warnings.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        f"Actual: {phishing_error_title}"
+    )
+
+    # Open a new tab
+    tabs.new_tab_by_button()
+    tabs.wait_for_num_tabs(2)
+    driver.switch_to.window(driver.window_handles[1])
+
+    # Test malware warning page
+    malware_url = "http://www.itisatrap.org/firefox/its-an-attack.html"
```
Comment (sarinali) [inline tests/security_and_privacy/test_phishing_and_malware_warnings.py:51]: here too!

**Review by sarinali — CHANGES_REQUESTED** [review]
Style! otherwise LGTM

File: tests/security_and_privacy/test_phishing_and_malware_warnings.py (hunk 4 above)
Comment (soncuteanca) [reply tests/security_and_privacy/test_phishing_and_malware_warnings.py]: Sure

**Review by emilghittasv — APPROVED** [review]
lgtm

**Review by ben-c-at-moz — APPROVED** [review]

**Review by sarinali — APPROVED** [review]
LGTM

**Pull request merged**